deploy/template.yaml

```diff
@@ -2317,7 +2347,10 @@ Resources:
       Period: 60 # This is the minimum value for the AWS Namespace
       EvaluationPeriods: 1
       DatapointsToAlarm: 1
-      Threshold: !FindInMap [ EnvironmentVariables, !Ref Environment, LAMBDACONCURRENCY80THRESHOLD ]
+      Threshold: !If
+        - ApplyReservedConcurrency
+        - !Ref LambdaConcurrencyThreshold
+        - !Ref AWS::NoValue
```

Comment:
Have we confirmed this will work when ApplyReservedConcurrency is false, as it would be trying to create an alarm with no specified Threshold value? It may be better to use the ApplyReservedConcurrency as a Condition on the whole alarm resource, since the alarm is not needed at all when reserved concurrencies are not applied. Thanks.